mapeditor/resourceExtractor/ResourceConverter.h

```diff
@@ -0,0 +1,49 @@
+/*
+ * ResourceConverter.h, part of VCMI engine
+ *
+ * Authors: listed in file AUTHORS in main folder
+ *
+ * License: GNU General Public License v2.0 or later
+ * Full text of license available in license.txt file, in main folder
+ *
+ */
+#pragma once
+
+namespace bfs = boost::filesystem;
+
+// Struct for holding all Convertor Options
+struct ConversionOptions
+{
+	bool splitDefs = false;			// splits TwCrPort, CPRSMALL, FlagPort, ITPA, ITPt, Un32 and Un44 into individual PNG's
+	bool convertPcxToPng = false;	// converts single Images (found in Images folder) from .pcx to png.
+	bool deleteOriginals = false;	// delete original files, for the ones splitted / converted.
+};
+
+// Struct for holding all Resource Extractor / Converter options
+struct ExtractionOptions
+{
+	bool extractArchives = false;	// if set, original H3 archives will be extracted into a separate folder
+	ConversionOptions conversionOptions;
+};
+
+class ResourceConverter
+{
+
+public:
+
+	// Splits def files that are shared between factions and converts pcx to bmp depending on Extraction Options
+	static void convertExtractedResourceFiles(ConversionOptions conversionOptions);
```

Comment:
Why just not make flags from ConversionOptions as fields of ResourceConverter?
Basically class methods are implicit calls of functions.
```
class Foo
{
    void bar();
};
```
is equivalent to 
```
class Foo {};
void bar(Foo *);

Reply:
I do not see ResourceConverter as a class with the functionality that it has right now. In the end it is just a simple function call. I grouped all functions in a static class and made only convertExtractedResourceFiles() public so that usage intention is clear.
I do not see the benefits of adding the fields to the class.